cpp/tensorrt_llm/cutlass_extensions/include/cutlass_extensions/gemm_configs.h

```diff
@@ -411,16 +413,17 @@ struct CutlassGemmConfig
     CutlassGemmConfig(CutlassTileConfigSM100 tile_config_sm100, MainloopScheduleType mainloop_schedule,
         EpilogueScheduleType epilogue_schedule, ClusterShape cluster_shape,
         ClusterShape dynamic_cluster_shape = ClusterShape::Undefined,
-        ClusterShape fallback_cluster_shape = ClusterShape::Undefined)
+        ClusterShape fallback_cluster_shape = ClusterShape::Undefined, int sm_version = 100)
         : tile_config_sm100(tile_config_sm100)
         , mainloop_schedule(mainloop_schedule)
         , epilogue_schedule(epilogue_schedule)
         , cluster_shape(cluster_shape)
         , dynamic_cluster_shape(dynamic_cluster_shape)
         , fallback_cluster_shape(fallback_cluster_shape)
-        , sm_version(100)
+        , sm_version(sm_version)
         , is_tma_warp_specialized(true)
     {
+        assert(sm_version >= 100 && sm_version < 120 && "Expected SM 10x version");
```

Comment:
the `assert` will only be triggered on DEBUG build, shall we use TLLM_CHECK_WITH_INFO?